crates/ironrdp-cliprdr-format/src/bitmap.rs

```diff
@@ -612,87 +610,83 @@ pub fn dib_to_png(input: &[u8]) -> Result<Vec<u8>, BitmapError> {
         return Err(BitmapError::Unsupported("unsupported compression"));
     }
 
-    let png_inputs = transform_bitmap(&header, src.remaining(), false)?;
-    encode_png(png_inputs)
+    let png_ctx = bgra_to_top_down_rgba(&header, src.remaining(), false)?;
+    encode_png(&png_ctx)
 }
 
-/// Convert `CF_DIB` to PNG.
+/// Converts `CF_DIB` to PNG.
 pub fn dibv5_to_png(input: &[u8]) -> Result<Vec<u8>, BitmapError> {
     let mut src = ReadCursor::new(input);
     let header = BitmapV5Header::decode(&mut src).map_err(BitmapError::InvalidHeader)?;
 
     validate_v5_header(&header)?;
 
-    let png_inputs = transform_bitmap(&header.header_v1, src.remaining(), true)?;
-    encode_png(png_inputs)
+    let png_ctx = bgra_to_top_down_rgba(&header.v1, src.remaining(), true)?;
+    encode_png(&png_ctx)
 }
 
-fn transform_png(info: png::OutputInfo, input_buffer: Vec<u8>) -> Result<(BitmapInfoHeader, Vec<u8>), BitmapError> {
+fn top_down_rgba_to_bottom_up_bgra(
+    info: png::OutputInfo,
+    src_bitmap: &[u8],
+) -> Result<(BitmapInfoHeader, Vec<u8>), BitmapError> {
```

Comment:
Similarly. Having "png" in the name misled me because at this point we expect the bitmap decoded from the PNG image, but we don’t actually decode and PNG format in this function.